test/functional/lua/json_spec.lua

```diff
@@ -152,6 +152,23 @@ describe('vim.json.encode()', function()
     clear()
   end)
 
+  it('dumps strings with escaped slash', function()
+    eq('"Test\\/"', exec_lua([[return vim.json.encode('Test/')]]))
+    eq('"Test\\/"', exec_lua([[return vim.json.encode('Test/', {})]]))
+    eq('"Test\\/"', exec_lua([[return vim.json.encode('Test/', { __invalid_key = true })]]))
+    eq('"Test\\/"', exec_lua([[return vim.json.encode('Test/', { escape_forward_slash = true })]]))
```

Comment:
probably makes sense to also test decode()?

    decode(encode(...))

Reply:
Added two cases for that, decode seems to be up to standard. It decodes `\\/` as `/`.